src/components/Counter/__tests__/counter.test.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import Counter from '../Counter';
+
+describe('Counter', () => {
+	it('renders initial value provided in props', () => {
+		const { getByText } = render(<Counter initialValue={5} />);
+		expect(getByText('5')).toBeInTheDocument();
```

Comment:
It works without issues if you don't have children components. 
In case you have children in component it is better to use `data-testid` attribute for each test component